pennylane/dla/lie_closure.py

```diff
@@ -24,6 +26,132 @@
 from pennylane.pauli import PauliSentence, PauliWord
 
 
+def lie_closure(
+    generators: Iterable[Union[PauliWord, PauliSentence, Operator]],
+    max_iterations: int = 10000,
+    verbose: int = 0,
+    pauli: bool = False,
+) -> Iterable[Union[PauliWord, PauliSentence, Operator]]:
+    r"""Compute the Lie closure over commutation of a set of generators
+
+    The Lie closure, pronounced "Lee" closure, is a way to compute the so-called dynamical Lie algebra (DLA) of a set of generators :math:`\mathcal{G} = \{G_1, .. , G_N\}`.
+    For such generators, one computes all nested commutators :math:`[G_i, [G_j, .., [G_k, G_\ell]]]` until no new operators are generated from commutation.
+    All these operators together form the DLA, see e.g. section IIB of `arXiv:2308.01432 <https://arxiv.org/abs/2308.01432>`__.
+
+    Args:
+        generators (Iterable[Union[PauliWord, PauliSentence, Operator]]): generating set for which to compute the
+            Lie closure.
+        max_iterations (int): maximum depth of nested commutators to consider. Default is ``10000``.
```

Comment:
How does the required number of iterations typically scale with the size of `generators` (number of qubits / size of generators set)?

Reply:
I am not aware of any general scaling with respect to the iterations. And I dont think there is one as it heavily depends on the topology and structure of the generators.

_Most_ final DLAs are exponential in the number of qubits.